sdk/src/asset_handlers/otf_io.rs

```diff
@@ -567,79 +755,98 @@ where
-            // create the other exclusion zone.
-            let offset = (&table_entry_buf[8..12]).read_u32::<BigEndian>()?;
-            let length = (&table_entry_buf[12..16]).read_u32::<BigEndian>()?;
-            positions.push(HashObjectPositions {
-                offset: offset as usize,
-                length: length as usize,
-                htype: HashBlockObjectType::Cai,
-            });
-        }
-        // Add the `head[checksumAdjustment]` value as it will change after
-        // adding C2PA
-        else if HEAD_TABLE_TAG.as_bytes() == table_tag {
-            let offset = (&table_entry_buf[8..12]).read_u32::<BigEndian>()?;
-            // Add in the `head`[checksumAdjustment] to the exclusions
-            positions.push(HashObjectPositions {
-                offset: offset as usize + 8,
-                length: 4,
-                htype: HashBlockObjectType::Cai,
-            });
-        }
-        table_counter += 1;
 
-        // If we have iterated over all of our tables, bail
-        if table_counter >= num_tables as usize {
-            break;
-        }
-    }
-    // Add the table directory headers
-    positions.push(HashObjectPositions {
-        offset: table_header_sz,
-        length: table_entry_sz * num_tables as usize,
-        htype: HashBlockObjectType::Cai,
-    });
-    for position in positions.iter().as_ref() {
-        debug!("Position for C2PA in font: {:?}", &position);
+    // Which will be built up from the different chunks from the file
+    let chunk_positions = otf_io.get_chunk_positions(&mut output_stream)?;
+    for chunk_position in chunk_positions {
+        let mut position_objs = match chunk_position.chunk_type {
+            // The table directory, other than the table records array will be
+            // added as "other"
+            ChunkType::TableDirectory => vec![HashObjectPositions {
+                offset: chunk_position.offset as usize,
+                length: chunk_position.length as usize,
+                htype: HashBlockObjectType::Other,
+            }],
+            // For the table record entries, we will specialize the C2PA table
+            // record and all others will be added as is
+            ChunkType::TableRecord => {
+                let table_record_pos = if &chunk_position.name == C2PA_TABLE_TAG.as_bytes() {
+                    vec![HashObjectPositions {
+                        offset: chunk_position.offset as usize,
+                        length: chunk_position.length as usize,
+                        htype: HashBlockObjectType::Cai,
+                    }]
+                } else {
+                    vec![HashObjectPositions {
+                        offset: chunk_position.offset as usize,
+                        length: chunk_position.length as usize,
+                        htype: HashBlockObjectType::Other,
+                    }]
+                };
+                table_record_pos
```

Comment:
I get an error here from VS Code (and github actions) but.. I don't really understand what the issue is.. haha

Reply:
Are you talking about clippy errors?